src/luxonis_ml/loader/augmentations.py

```diff
@@ -33,82 +49,219 @@ def _parse_cfg(
+                if aug["name"] in ["Mosaic4"]:
+                    self.is_batched = True
+                    self.batch_size = 4
+                    multi_image_augs.append(eval(aug["name"])(**aug.get("params", {})))
+                else:
+                    single_image_augs.append(eval(aug["name"])(**aug.get("params", {})))
+
+        batch_transform = A.BatchCompose(
+            [
+                A.ForEach(single_image_augs),
+                *multi_image_augs,
+            ],
+            bbox_params=A.BboxParams(format="coco", label_fields=["bboxes_classes_batch"]),
             keypoint_params=A.KeypointParams(
-                format="xy", label_fields=["keypoints_classes"], remove_invisible=False
+                format="xy", 
+                label_fields=["keypoints_visibility_batch", "keypoints_classes_batch"],
+                remove_invisible=False
             ),
         )
+        return batch_transform, default_transform
+
+
+    def __call__(self, data: List[Tuple[np.ndarray, dict]]):
+        """Performs augmentations on provided data
+
+        Args:
+            data (List[Tuple[np.ndarray, dict]]): Data with list of input images and their annotations
+
+        Returns:
+            Tuple[np.ndarray, dict]: Transformed output image and its annotations
+        """
+        image_batch = []
+        mask_batch = []
+        bboxes_batch = []
+        bboxes_classes_batch = []
+        keypoints_batch = []
+        keypoints_visibility_batch = []
+        keypoints_classes_batch = []
+
+        present_annotations = set()
+        for img, annotations in data:
+            present_annotations.update(annotations.keys())
+            classes, mask, bboxes_points, bboxes_classes, keypoints_points, \
+            keypoints_visibility, keypoints_classes, n_kpts_per_instance = self.prepare_img_annotations(annotations, *img.shape[:-1])
+
+            image_batch.append(img)
+            mask_batch.append(mask)
+            bboxes_batch.append(bboxes_points)
+            bboxes_classes_batch.append(bboxes_classes)
+            keypoints_batch.append(keypoints_points)
+            keypoints_visibility_batch.append(keypoints_visibility)
+            keypoints_classes_batch.append(keypoints_classes)
+
+        # Apply transforms
+        # NOTE: All keys (including label_fields) must have _batch suffix when using BatchCompose
+        transformed = self.batch_transform(
+            image_batch=image_batch,
+            mask_batch=mask_batch,
+            bboxes_batch=bboxes_batch,
+            bboxes_classes_batch=bboxes_classes_batch,
+            keypoints_batch=keypoints_batch,
+            keypoints_visibility_batch=keypoints_visibility_batch,
+            keypoints_classes_batch=keypoints_classes_batch
+        )
+
+        # convert to numpy arrays
+        for key in transformed:
+            transformed[key] = np.array(transformed[key][0])
+
+        transformed = self.default_transform(
```

Comment:
I meant to comment on this earlier but since `default_transform` seems to happen last, but we automatically adjust the resize to be larger in the case of `Mosaic4`? For example, we'd want width and height x2 compared to the normal training size. And the user might not remember to adjust this manually in the training config.

Reply:
There are `out_height `and `out_width ` parameters for `Mosaic4 `that have to be defined by the user ([this](https://github.com/i-aki-y/albumentations/blob/c137e2b8bb48eb25b8865db80027f16059724db7/albumentations/augmentations/geometric/transforms.py#L1510-L1515)). So if we would want bigger crops of images in each quadrant this two values should be high (e.g. for full image in each quadrant this should be 2xH by 2xW). But this only works if all images have same size (not true in COCO for example). 

Do we want to automatically x2 shape if this parameters are not set by the user? And if so, where should we handle this: in Augmentations or before, when loading config in `models` for example?